src/resolve.js

```diff
@@ -52,13 +52,19 @@ const createResolve = (config) => {
 
   const cache = {}
 
-  const resolve = (filename) => {
+  const resolve = (filename, contentFromFile) => {
     if (!cache[filename]) {
-      const content = readFileSync(filename).toString()
+      // contentFromFile is here for testing purposes only
+      const content = contentFromFile || readFileSync(filename).toString()
```

Comment:
every mock I've tried for `fs.readFileSync` inevitably broke files from functions from the `resolve` module.
this is the only solution that worked.